### PR TITLE
Space cobra

### PR DESCRIPTION
**Pull request by Nimfar11**
## About the PR

NT scanners have spotted a new species of space critter, they are invisible most of the time and have blind poison in their fangs. The NT requires all employees in the wreckage to be extremely cautious.

**Media**

cobra

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

:cl: Nimfar11
- add: Adds Space Cobras, invisible at rest and venom blind.
- add: Adds snake meat and fancy snakeskin boots.

### CODE REVIEW

**Comment by Nimfar11** [comment]
I found mistakes, I'll fix them soon

**Comment by Nimfar11** [comment]
There are still problems with invisibility and the inability to find the creature at the time of death. Solve the problem in time

File: Resources/Prototypes/Entities/Mobs/NPCs/space.yml [hunk 1]
```diff
@@ -301,3 +301,109 @@
...
+    excess: 150
+  - type: DamageStateVisuals
+    states:
+      Alive:
+        Base: space_cobra
+        BaseUnshaded: glow
+      Critical:
+        Base: dead_space_cobra
+      Dead:
+        Base: dead_space_cobra
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/NPCs/space.yml:338]: Mobs shouldn't have crit states.

**Comment by Nimfar11** [comment]
> just remove stealth ocmp on crit

Yeah, I thought of that too, but you have to get into the stealth system code, and I'm not very good with code. That's why it takes time. :)

**Comment by deltanedas** [comment]
i think stealth system handles stealth component removal -> stop being blue + invisible so just RemComp<StealthComponent>

File: Resources/Prototypes/Entities/Mobs/NPCs/space.yml (hunk 1 above)
Comment (metalgearsloth) [reply Resources/Prototypes/Entities/Mobs/NPCs/space.yml:338]: Yeah, the other ones need removing too at some point.

Comment (Nimfar11) [reply Resources/Prototypes/Entities/Mobs/NPCs/space.yml]: Removed the crits from space mobs

**Comment by Nimfar11** [comment]
It worked, invisibility is turned off when a mob dies

File: Content.Shared/Stealth/SharedStealthSystem.cs [hunk 2]
```diff
@@ -22,6 +25,7 @@ public override void Initialize()
         SubscribeLocalEvent<StealthComponent, ComponentInit>(OnInit);
         SubscribeLocalEvent<StealthComponent, ExamineAttemptEvent>(OnExamineAttempt);
         SubscribeLocalEvent<StealthComponent, ExaminedEvent>(OnExamined);
+		SubscribeLocalEvent<StealthComponent, MobStateChangedEvent>(OnMobStateChanged);
```
Comment (metalgearsloth) [inline Content.Shared/Stealth/SharedStealthSystem.cs:28]: Looks like tabs have been mixed in where spaces should always be used.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Shared/Stealth/SharedStealthSystem.cs (hunk 2 above)
Comment (Nimfar11) [reply Content.Shared/Stealth/SharedStealthSystem.cs]: done

**Comment by metalgearsloth** [comment]
Tests still need fixing.

File: Resources/Audio/Effects/Footsteps/attributions.yml [hunk 3]
```diff
@@ -25,3 +25,11 @@
   license: "CC-BY-SA-3.0"
   copyright: "Taken from https://github.com/tgstation/tgstation"
   source: "https://github.com/nero1024/tgstation/blob/83ccc939a20489de8ab81cb47e6f8e84c490adc2/sound/effects/footstep/slime1.ogg"
+
+- files:
+  - snake1.ogg
+  - snake2.ogg
+  - snake3.ogg
+  license: "CC-BY-SA-3.0"
```
Comment (ficcialfaint) [inline Resources/Audio/Effects/Footsteps/attributions.yml:33]: The site you took sounds from doesn’t say that these sounds are CC-BY-SA-3.0

Comment (Nimfar11) [reply Resources/Audio/Effects/Footsteps/attributions.yml:33]: It says they are completely free. So I have to delete the license?

Comment (metalgearsloth) [reply Resources/Audio/Effects/Footsteps/attributions.yml:33]: Free is not CC-BY-SA-3.0 which is why freesound.org is easier.

**Comment by Nimfar11** [comment]
I need some advice, I don't know what else to do here.

File: Resources/Audio/Effects/Footsteps/attributions.yml (hunk 3 above)
Comment (Nimfar11) [reply Resources/Audio/Effects/Footsteps/attributions.yml:33]: It says on their website that they only require that their website be listed as the source. Which is accomplished here.

Comment (metalgearsloth) [reply Resources/Audio/Effects/Footsteps/attributions.yml:33]: Just go with "Custom" and add the stipulation.

**Comment by metalgearsloth** [comment]
Re-open when you get a chance to fix this.

**Pull request closed without merging**